src/main/java/uk/ac/ebi/atlas/experimentpage/markergenes/MarkerGeneServiceImpl.java

```diff
@@ -20,8 +22,14 @@ public MarkerGeneServiceImpl(MarkerGenesDao markerGenesDao) {
      * @return ImmutableList of CellTypeMarkerGene Objects
      */
     @Override
-    public ImmutableList<CellTypeMarkerGene> getCellTypeMarkerGeneProfile(String experimentAccession, String organismPart) {
-        List<CellTypeMarkerGene> cellTypeMarkerGenes = markerGenesDao.getCellTypeMarkerGenes(experimentAccession, organismPart);
-        return ImmutableList.copyOf(cellTypeMarkerGenes);
+    public Map<String, ImmutableSet<CellTypeMarkerGene>> getCellTypeMarkerGeneProfile(String experimentAccession, String organismPart) {
+        return markerGenesDao.getCellTypeMarkerGenes(experimentAccession, organismPart)
+                .stream()
+                .collect(groupingBy(CellTypeMarkerGene::cellGroupValue, toImmutableSet())).entrySet().stream()
+                .collect(toMap(cellTypeMarkerGenes -> cellTypeMarkerGenes.getKey(),
```

Comment:
Use `toImmutableMap` and return type accordingly.

Reply:
Fixed.

Reply:
I will add this missing test case for the top5 genes in the [Update cluster PR](https://github.com/ebi-gene-expression-group/atlas-web-single-cell/pull/113).